pkg/block/index.go

```diff
@@ -239,22 +290,22 @@ func GatherIndexIssueStats(fn string, minTime int64, maxTime int64) (stats Index
 		}
 
 		ooo := 0
-		if chks[0].MinTime < minTime || chks[0].MaxTime > maxTime {
-			stats.Outsiders++
-			if chks[0].MinTime > maxTime || chks[0].MaxTime < minTime {
-				stats.CompleteOutsiders++
-			}
-		}
-		for i, c := range chks[1:] {
-			c0 := chks[i]
-
+		// Per chunk in series.
+		for i, c := range chks {
 			if c.MinTime < minTime || c.MaxTime > maxTime {
-				stats.Outsiders++
+				stats.OutsideChunks++
 				if c.MinTime > maxTime || c.MaxTime < minTime {
-					stats.CompleteOutsiders++
+					stats.CompleteOutsideChunks++
+				} else if c.MinTime == maxTime && c.MaxTime > maxTime {
```

Comment:
I don't know if it can actually happen, but in theory a chunk with a single value could have `c.MinTime == c.MaxTime`, and if it's also at the very end of the block (`c.MinTime == maxTime`), it wouldn't be picked up by this condition.

I think you just need to test `if c.MinTime == maxTime`.

Reply:
Can agree with it, but single sample chunk is very sad chunk (: But can happen probably.